frontend/amundsen_application/static/js/features/ColumnList/ColumnDetailsPanel/index.tsx

```diff
@@ -0,0 +1,131 @@
+// Copyright Contributors to the Amundsen project.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as React from 'react';
+import EditableSection from 'components/EditableSection';
+import BadgeList from 'features/BadgeList';
+import ColumnDescEditableText from 'features/ColumnList/ColumnDescEditableText';
+import ColumnLineage from 'features/ColumnList/ColumnLineage';
+import ColumnStats from 'features/ColumnList/ColumnStats';
+import ExpandableUniqueValues from 'features/ExpandableUniqueValues';
+import { getMaxLength, isColumnListLineageEnabled } from 'config/config-utils';
+import { getColumnLink } from 'utils/navigationUtils';
+import { filterOutUniqueValues, getUniqueValues } from 'utils/stats';
+import { FormattedDataType } from '..';
+import {
+  COPY_COL_LINK_LABEL,
+  COPY_COL_NAME_LABEL,
+  EDITABLE_SECTION_TITLE,
+  CLOSE_LABEL,
+} from './constants';
+
+export interface ColumnDetailsPanelProps {
+  columnDetails: FormattedDataType;
+  togglePanel: (columnDetails: FormattedDataType, event: any) => void;
+}
+
+const shouldRenderDescription = (columnDetails: FormattedDataType) => {
+  const { content, editText, editUrl, isEditable } = columnDetails;
+  if (content.description) {
+    return true;
+  }
+  if (!editText && !editUrl && !isEditable) {
+    return false;
+  }
+  return true;
+};
+
+const ColumnDetailsPanel: React.FC<ColumnDetailsPanelProps> = ({
+  columnDetails,
+  togglePanel,
+}: ColumnDetailsPanelProps) => {
+  const {
+    content,
+    stats,
+    editText,
+    editUrl,
+    col_index,
+    name,
+    tableParams,
+    isEditable,
+    badges,
+  } = columnDetails;
+
+  const normalStats = stats && filterOutUniqueValues(stats);
+  const uniqueValueStats = stats && getUniqueValues(stats);
+
+  return (
+    <aside className="right-panel">
+      <div className="panel-header">
+        <div className="panel-title">{name}</div>
```

Comment:
This should be a header probably, not sure which level (maybe h3?). We don't want to render text on div tags, as they are not semantic.